scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,680 @@
+#!/usr/bin/env bash -x
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="skyscraper"
+rp_module_desc="Scraper for EmulationStation by Lars Muldjord"
+rp_module_licence="GPLv3.0 https://raw.githubusercontent.com/muldjord/skyscraper/master/LICENSE"
+rp_module_section="opt"
+
+function depends_skyscraper() {
+    getDepends qt5-default p7zip-full
+}
+
+function sources_skyscraper() {
+    local latest_ver
+    latest_ver=$(latest_ver_skyscraper)
+
+    gitPullOrClone "$md_build" "https://github.com/muldjord/skyscraper" "$latest_ver"
+}
+
+function build_skyscraper() {
+    qmake
+    make
+    md_ret_require="$md_build/Skyscraper"
+}
+
+function install_skyscraper() {
+    md_ret_files=(
+    'Skyscraper'
+    'LICENSE'
+    'README.md'
+    )
+}
+
+# Purge all Skyscraper caches
+function _purge_skyscraper() {
+    find "$configdir/all/skyscraper/dbs" -maxdepth 1 -mindepth 1 -type d -print0 -exec rm -fr {} \;
+}
+
+# Purge by cached platform
+function _purge_platform_skyscraper() {
+    local options=()
+    local system
+
+    while read system; do
+        # Get the size on disk of the system and show it in the select list
+        local size=$(du -sh  "$configdir/all/skyscraper/dbs/$system" | cut -f1)
+        options+=("$system" "$size" OFF)
+    done< <(find "$configdir/all/skyscraper/dbs" -maxdepth 1 -mindepth 1 -type d -exec basename {} \;)
+
+    # If not folders are found, show an info message instead of the selection list
+    if [[ ${#options[@]} -eq 0 ]] ; then
+        printMsgs "dialog" "Nothing to delete ! No folders found in \n$configdir/all/skyscraper/dbs "
+        return
+    fi
+
+    local cmd=(dialog --backtitle "$__backtitle" --radiolist "Select platform to purge" 20 60 12)
+    platform=$("${cmd[@]}" "${options[@]}" 2>&1 >/dev/tty)
+
+    # Exit if no platform chosen
+    [[ -z "$platform" ]] && return
+
+    [[ -d "$configdir/all/skyscraper/dbs/$platform" ]] && rm -fr "$configdir/all/skyscraper/dbs/$platform"
+
+}
+
+
+function get_ver_skyscraper() {
+    [[ -f "$md_inst/Skyscraper" ]] && "$md_inst/Skyscraper" -h | grep 'Running Skyscraper'  | cut -d' '  -f 3 2>/dev/null
+}
+
+function latest_ver_skyscraper() {
+    # Prefer `wget` to `curl`, sinc it's already installed and used by the RetroPie script
+    wget -qO- https://api.github.com/repos/muldjord/skyscraper/releases/latest | grep -m 1 tag_name | cut -d\" -f4
+}
```

Comment:
is this function needed to be called manually by users ? if not then it should be hidden with _ prefix from retropie_packages.

Reply:
No, I'll hide it.

Reply:
thanks. might be worth checking other functions also - eg get_ver